imap/caldav_alarm.c

```diff
@@ -1359,6 +1426,51 @@ static int process_futurerelease(struct mailbox *mailbox,
 
     caldav_alarm_delete_record(mailbox_name(mailbox), record->uid);
 
+    onSend = json_object_get(submission, "onSend");
+    if (onSend) {
+        json_t *emailid = json_object_get(submission, "emailId");
```

Comment:
Better to pull the string value out of the JSON value here, and for destmboxid and setkeywords later? This would also allow to better handle NULL values in the "that-never-happens" case.